pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -79,13 +119,14 @@ func (pc *PodCache) onEvent(curr interface{}, ev model.Event) error {
 			return fmt.Errorf("tombstone contained object that is not a pod %#v", curr)
 		}
 	}
-
+	inTransitState := false
```

Comment:
This is not clear to me without reading PR comments, can you add comments in  the code?